independent-projects/parent/pom.xml

```diff
@@ -0,0 +1,434 @@
+        <maven.compiler.parameters>true</maven.compiler.parameters>
+        <maven.compiler.release>11</maven.compiler.release>
+        <maven.compiler.source>${maven.compiler.release}</maven.compiler.source>
+        <maven.compiler.target>${maven.compiler.release}</maven.compiler.target>
+        <maven.compiler.testTarget>${maven.compiler.target}</maven.compiler.testTarget>
+        <maven.compiler.testSource>${maven.compiler.source}</maven.compiler.testSource>
+        <maven.javadoc.failOnError>false</maven.javadoc.failOnError>
+
+        <!-- Cross plugins settings -->
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+        <!--
+            Options to override the compiler arguments directly on the compiler argument line to separate between what
+            the IDE understands as the source level and what the Maven compiler actually use.
+        -->
+        <maven.compiler.argument.target>${maven.compiler.target}</maven.compiler.argument.target>
+        <maven.compiler.argument.source>${maven.compiler.source}</maven.compiler.argument.source>
+        <maven.compiler.argument.testTarget>${maven.compiler.testTarget}</maven.compiler.argument.testTarget>
+        <maven.compiler.argument.testSource>${maven.compiler.testSource}</maven.compiler.argument.testSource>
+
+        <!-- maven-enforcer-plugin -->
+        <maven.min.version>3.8.6</maven.min.version>
+        <jdk.min.version>${maven.compiler.argument.source}</jdk.min.version>
+        <insecure.repositories>ERROR</insecure.repositories>
+
+        <!-- maven-release-plugin -->
+        <useReleaseProfile>false</useReleaseProfile>
+    </properties>
+
+    <licenses>
+        <license>
+            <name>Apache License, Version 2.0</name>
+            <url>https://www.apache.org/licenses/LICENSE-2.0.txt</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+
+    <issueManagement>
+        <system>GitHub</system>
+        <url>https://github.com/quarkusio/quarkus/issues</url>
+    </issueManagement>
+
+    <developers>
+        <developer>
+            <id>quarkus</id>
+            <name>Quarkus Community</name>
+        </developer>
+    </developers>
+
+    <scm>
+        <connection>scm:git:git@github.com:quarkusio/quarkus.git</connection>
+        <developerConnection>scm:git:git@github.com:quarkusio/quarkus.git</developerConnection>
+        <url>https://github.com/quarkusio/quarkus</url>
+        <tag>HEAD</tag>
+    </scm>
+
+    <distributionManagement>
+        <snapshotRepository>
+            <id>ossrh</id>
+            <url>https://s01.oss.sonatype.org/content/repositories/snapshots/</url>
+        </snapshotRepository>
+        <repository>
+            <id>oss.sonatype</id>
+            <url>https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
```

Comment:
> This is just fyi as other people may hit the similar challenges.
> 
> I noticed one side effect after this change, upload of bits into our internal artifactory fails with `authentication failed for http://10.0.139.54:8040/artifactory/libs-snapshot-local/io/quarkus/quarkus-parent/999-SNAPSHOT/quarkus-parent-999-20230524.072939-1034.pom, status: 401 Unauthorized -> [Help 1]`
> 
> No changes on the instance side, uploads were working before this change, the diff is in the plugin version
> 
> ```
> [INFO] --- maven-deploy-plugin:2.8.2:deploy (default-deploy) @ quarkus-parent ---    => OK
> [INFO] --- maven-deploy-plugin:3.1.1:deploy (default-deploy) @ quarkus-parent ---    => FAILS
> ```
> 
> I will investigate it further next week, I'm on PTO starting tmr and my day is booked already.

I took a look at this and here's what I've learned so far.

We do the upload to our internal artifactory by locally modifying the `distributionManagement` in Quarkus' POMs:
- Replace Sonatype's repo URLs with our internal URLs.
- Replace the repo IDs with our own IDs, which correspond to our custom `settings.xml` containing internal credentials.
```bash
for i in `git grep -l oss.sonatype.org/content | grep xml`; do
sed -i "s,https://s01.oss.sonatype.org/content/repositories/snapshots,http://10.0.139.54:8040/artifactory/libs-snapshot-local,g" $i
sed -i "s,https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/,http://10.0.139.54:8040/artifactory/libs-release-local,g" $i
sed -i "s,sonatype-nexus-snapshots,artifactory-snapshots,g" $i
sed -i "s,sonatype-nexus-release,artifactory-release,g" $i
done
```

The `distributionManagement` in parent POM gets messed up by this, as it uses different IDs.

An obvious action on our side in order to fix our upload would be to modify the replacement script so that it takes the different IDs into account.

However, I still don't understand why the repo IDs in parent POM are different from IDs in every other POM. Could pls you shed some light on it? @gastaldi @gsmet 

cc @rsvoboda

Reply:
@jsmrcka could you point me to pom files with different ids in our tree? I agree we should make them consistent.

Reply:
OK, I will fix it then.